dvc_http/__init__.py

```diff
@@ -129,28 +128,17 @@ async def get_client(self, **kwargs):
 
     @cached_property
     def fs(self):
-        from fsspec.implementations.http import (
-            HTTPFileSystem as _HTTPFileSystem,
-        )
+        from .spec import HTTPFileSystemWrapper
 
-        return _HTTPFileSystem(get_client=self.get_client, **self.fs_args)
+        return HTTPFileSystemWrapper(
+            get_client=self.get_client,
+            upload_method=self.upload_method,
```

Comment:
Let's get rid of `self.upload_method` and set these in `__init__()` itself in `self.fs_args`.

```python
self.fs_args["upload_method"] = kwargs.get("method", "POST")
```